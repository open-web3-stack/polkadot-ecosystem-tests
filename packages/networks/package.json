{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
