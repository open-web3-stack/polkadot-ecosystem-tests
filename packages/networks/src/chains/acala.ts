import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
