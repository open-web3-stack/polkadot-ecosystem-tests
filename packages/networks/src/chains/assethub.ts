import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
