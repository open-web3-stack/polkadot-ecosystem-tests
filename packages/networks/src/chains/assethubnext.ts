import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
