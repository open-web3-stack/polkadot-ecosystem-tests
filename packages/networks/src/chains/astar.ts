import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
