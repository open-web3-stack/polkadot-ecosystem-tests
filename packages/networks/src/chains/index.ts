export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
