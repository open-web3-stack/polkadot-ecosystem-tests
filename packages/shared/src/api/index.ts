import type { ApiPromise } from '@polkadot/api'

export type TransferType = 'Teleport' | 'LocalReserve' | 'DestinationReserve' | 'RemoteReserve'

export const xtokens = {
  relaychainV3: (acc: any) => ({
    V3: {
      parents: 1,
      interior: {
        X1: {
          AccountId32: {
            id: acc,
          },
        },
      },
    },
  }),
  relaychainV4: (acc: any) => ({
    V4: {
      parents: 1,
      interior: {
        X1: {
          AccountId32: {
            id: acc,
          },
        },
      },
    },
  }),
  parachainAccountId20V3: (paraId: number) => (acc: any) => ({
    V3: {
      parents: 1,
      interior: {
        X2: [
          { Parachain: paraId },
          {
            AccountKey20: {
              key: acc,
            },
          },
        ],
      },
    },
  }),
  parachainV3: (paraId: number) => (acc: any) => ({
    V3: {
      parents: 1,
      interior: {
        X2: [
          { Parachain: paraId },
          {
            AccountId32: {
              id: acc,
            },
          },
        ],
      },
    },
  }),
  parachainV4: (paraId: number) => (acc: any) => ({
    V4: {
      parents: 1,
      interior: {
        X2: [
          { Parachain: paraId },
          {
            AccountId32: {
              id: acc,
            },
          },
        ],
      },
    },
  }),
  transfer:
    (token: any, amount: any, dest: (dest: any) => any, weight: any = 'Unlimited') =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      api.tx.xTokens.transfer(token, amount, dest(acc), weight),
  transferMulticurrencies:
    (token: any, amount: any, feeToken: any, feeAmount: any, dest: (dest: any) => any) =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      api.tx.xTokens.transferMulticurrencies(
        [
          [token, amount],
          [feeToken, feeAmount],
        ],
        1,
        dest(acc),
        'Unlimited',
      ),
}

export const xcmPallet = {
  relaychainV4: {
    V4: {
      parents: 1,
      interior: 'Here',
    },
  },
  parachainV3: (parents: number, paraId: any) => ({
    V3: {
      parents,
      interior: {
        X1: { Parachain: paraId },
      },
    },
  }),
  parachainV4: (parents: number, paraId: any) => ({
    V4: {
      parents,
      interior: {
        X1: [{ Parachain: paraId }],
      },
    },
  }),
  limitedTeleportAssets:
    (token: any, amount: any, dest: any) =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      (api.tx.xcmPallet || api.tx.polkadotXcm).limitedTeleportAssets(
        dest,
        {
          V3: {
            parents: 0,
            interior: {
              X1: {
                AccountId32: {
                  // network: 'Any',
                  id: acc,
                },
              },
            },
          },
        },
        {
          V3: [
            {
              id: token,
              fun: { Fungible: amount },
            },
          ],
        },
        0,
        'Unlimited',
      ),
  limitedReserveTransferAssetsV3:
    (token: any, amount: any, dest: any) =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      (api.tx.xcmPallet || api.tx.polkadotXcm).limitedReserveTransferAssets(
        dest,
        {
          V3: {
            parents: 0,
            interior: {
              X1: {
                AccountId32: {
                  id: acc,
                },
              },
            },
          },
        },
        {
          V3: [
            {
              id: token,
              fun: { Fungible: amount },
            },
          ],
        },
        0,
        'Unlimited',
      ),
  teleportAssetsV3:
    (token: any, amount: any, dest: any) =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      (api.tx.xcmPallet || api.tx.polkadotXcm).teleportAssets(
        dest,
        {
          V3: {
            parents: 0,
            interior: {
              X1: {
                AccountId32: {
                  id: acc,
                },
              },
            },
          },
        },
        {
          V3: [
            {
              id: token,
              fun: { Fungible: amount },
            },
          ],
        },
        0,
      ),
  executeXCM:
    (xcm: any, max_weight: any) =>
    ({ api }: { api: ApiPromise }) => {
      return (api.tx.xcmPallet || api.tx.polkadotXcm).execute(xcm, max_weight)
    },
  transferAssetsV3:
    (token: any, amount: any, dest: any) =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      (api.tx.xcmPallet || api.tx.polkadotXcm).transferAssets(
        dest,
        {
          V3: {
            parents: 0,
            interior: {
              X1: {
                AccountId32: {
                  id: acc,
                },
              },
            },
          },
        },
        {
          V3: [
            {
              id: token,
              fun: { Fungible: amount },
            },
          ],
        },
        0,
        'Unlimited',
      ),
  transferAssetsUsingType:
    (dest: any, tokens: any[], assetTransferType: TransferType, remoteFeesId: any, feesTransferType: TransferType) =>
    ({ api }: { api: ApiPromise }, acc: any) =>
      (api.tx.xcmPallet || api.tx.polkadotXcm).transferAssetsUsingTypeAndThen(
        dest,
        { V4: tokens },
        assetTransferType,
        { V4: remoteFeesId },
        feesTransferType,
        {
          V4: [
            {
              DepositAsset: {
                assets: {
                  Wild: {
                    AllCounted: 2,
                  },
                },
                beneficiary: {
                  parents: 0,
                  interior: { X1: [{ AccountId32: { id: acc } }] },
                },
              },
            },
          ],
        },
        'Unlimited',
      ),
}

export const tx = {
  xtokens,
  xcmPallet,
}

export const query = {
  balances: ({ api }: { api: ApiPromise }, address: string) => api.query.system.account(address),
  totalIssuance: ({ api }: { api: ApiPromise }) => api.query.balances.totalIssuance(),
  tokens:
    (token: any) =>
    ({ api }: { api: ApiPromise }, address: string) =>
      api.query.tokens.accounts(address, token),
  assets:
    (token: number | bigint) =>
    ({ api }: { api: ApiPromise }, address: string) =>
      api.query.assets.account(token, address),
  foreignAssets:
    (token: any) =>
    ({ api }: { api: ApiPromise }, address: string) =>
      api.query.foreignAssets.account(token, address),
  evm:
    (contract: string, slot: string) =>
    ({ api }: { api: ApiPromise }, _address: string) =>
      api.query.evm.accountStorages(contract, slot),
}
