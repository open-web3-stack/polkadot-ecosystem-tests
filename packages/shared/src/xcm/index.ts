export * from './runXcmPalletDown.js'
export * from './runXcmPalletHorizontal.js'
export * from './runXcmPalletUp.js'
export * from './runXtokenstHorizontal.js'
export * from './runXtokensUp.js'
