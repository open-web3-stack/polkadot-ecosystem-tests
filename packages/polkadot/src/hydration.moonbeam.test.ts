import { describe } from 'vitest'

import { defaultAccounts } from '@e2e-test/networks'
import { hydration, moonbeam, polkadot } from '@e2e-test/networks/chains'
import { setupNetworks } from '@e2e-test/shared'
import { query, tx } from '@e2e-test/shared/api'
import { runXcmPalletHorizontal, runXtokenstHorizontal } from '@e2e-test/shared/xcm'

describe('hydration & moonbeam', async () => {
  const [hydrationClient, moonbeamClient, polkadotClient] = await setupNetworks(hydration, moonbeam, polkadot)

  const hydrationDot = hydration.custom.relayToken
  const moonbeamDot = moonbeam.custom.dot
  const glmr = hydration.custom.glmr

  runXtokenstHorizontal('hydration transfer DOT to moonbeam', async () => {
    return {
      fromChain: hydrationClient,
      fromBalance: query.tokens(hydrationDot),
      fromAccount: defaultAccounts.alice,

      toChain: moonbeamClient,
      toBalance: query.assets(moonbeamDot),
      toAccount: defaultAccounts.alith,

      routeChain: polkadotClient,
      isCheckUmp: true,

      tx: tx.xtokens.transfer(hydrationDot, 2e12, tx.xtokens.parachainAccountId20V3(moonbeam.paraId!)),
    }
  })

  runXcmPalletHorizontal('moonbeam transfer DOT to hydration', async () => {
    await moonbeamClient.dev.setStorage({
      Assets: {
        account: [[[moonbeamDot, defaultAccounts.alith.address], { balance: 10e12 }]],
      },
    })

    return {
      fromChain: moonbeamClient,
      fromBalance: query.assets(moonbeamDot),
      fromAccount: defaultAccounts.alith,

      toChain: hydrationClient,
      toBalance: query.tokens(hydrationDot),
      toAccount: defaultAccounts.bob,

      routeChain: polkadotClient,
      isCheckUmp: true,

      tx: tx.xcmPallet.transferAssetsV3(moonbeam.custom.xcmDot, 2e10, tx.xcmPallet.parachainV3(1, hydration.paraId!)),
    }
  })

  runXtokenstHorizontal('hydration transfer GLMR to moonbeam', async () => {
    await hydrationClient.dev.setStorage({
      Tokens: {
        Accounts: [[[defaultAccounts.alice.address, glmr], { free: '50000000000000000000000' }]],
      },
    })

    return {
      fromChain: hydrationClient,
      fromBalance: query.tokens(glmr),
      fromAccount: defaultAccounts.alice,
      routeChain: polkadotClient,
      toChain: moonbeamClient,
      toBalance: query.balances,
      toAccount: defaultAccounts.baltathar,

      tx: tx.xtokens.transfer(glmr, '200000000000000000000', tx.xtokens.parachainAccountId20V3(moonbeam.paraId!)),
    }
  })
})
