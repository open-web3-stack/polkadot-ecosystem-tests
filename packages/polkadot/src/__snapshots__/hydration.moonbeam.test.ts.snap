// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`hydration & moonbeam > hydration transfer DOT to moonbeam > balance on from chain 1`] = `
{
  "free": 998000000000000,
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`hydration & moonbeam > hydration transfer DOT to moonbeam > balance on to chain 1`] = `
{
  "balance": "(rounded 2000000000000)",
  "extra": null,
  "reason": {
    "sufficient": null,
  },
  "status": "Liquid",
}
`;

exports[`hydration & moonbeam > hydration transfer DOT to moonbeam > from chain ump messages 1`] = `
[
  {
    "v3": [
      {
        "withdrawAsset": [
          {
            "fun": {
              "fungible": 2000000000000,
            },
            "id": {
              "concrete": {
                "interior": {
                  "here": null,
                },
                "parents": 0,
              },
            },
          },
        ],
      },
      {
        "clearOrigin": null,
      },
      {
        "buyExecution": {
          "fees": {
            "fun": {
              "fungible": 1000000000000,
            },
            "id": {
              "concrete": {
                "interior": {
                  "here": null,
                },
                "parents": 0,
              },
            },
          },
          "weightLimit": {
            "unlimited": null,
          },
        },
      },
      {
        "depositReserveAsset": {
          "assets": {
            "wild": {
              "allCounted": 1,
            },
          },
          "dest": {
            "interior": {
              "x1": {
                "parachain": 2004,
              },
            },
            "parents": 0,
          },
          "xcm": [
            {
              "buyExecution": {
                "fees": {
                  "fun": {
                    "fungible": 1000000000000,
                  },
                  "id": {
                    "concrete": {
                      "interior": {
                        "here": null,
                      },
                      "parents": 1,
                    },
                  },
                },
                "weightLimit": {
                  "unlimited": null,
                },
              },
            },
            {
              "depositAsset": {
                "assets": {
                  "wild": {
                    "allCounted": 1,
                  },
                },
                "beneficiary": {
                  "interior": {
                    "x1": {
                      "accountKey20": {
                        "key": "0xf24ff3a9cf04c71dbc94d0b566f7a27b94566cac",
                        "network": null,
                      },
                    },
                  },
                  "parents": 0,
                },
              },
            },
          ],
        },
      },
      {
        "setTopic": "(redacted)",
      },
    ],
  },
]
`;

exports[`hydration & moonbeam > hydration transfer DOT to moonbeam > route chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": {
        "Ump": {
          "Para": "(rounded 2000)",
        },
      },
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 7200)",
        "refTime": "(rounded 580000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`hydration & moonbeam > hydration transfer DOT to moonbeam > to chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": "Parent",
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 29000)",
        "refTime": "(rounded 4200000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`hydration & moonbeam > hydration transfer DOT to moonbeam > tx events 1`] = `
[
  {
    "data": {
      "assets": [
        {
          "fun": {
            "Fungible": 2000000000000,
          },
          "id": {
            "interior": "Here",
            "parents": 1,
          },
        },
      ],
      "dest": {
        "interior": {
          "X2": [
            {
              "Parachain": "(rounded 2000)",
            },
            {
              "AccountKey20": {
                "key": "0xf24ff3a9cf04c71dbc94d0b566f7a27b94566cac",
                "network": null,
              },
            },
          ],
        },
        "parents": 1,
      },
      "fee": {
        "fun": {
          "Fungible": 2000000000000,
        },
        "id": {
          "interior": "Here",
          "parents": 1,
        },
      },
      "sender": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "TransferredAssets",
    "section": "xTokens",
  },
]
`;

exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > balance on from chain 1`] = `
{
  "free": 4.98e+22,
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > balance on to chain 1`] = `
{
  "consumers": 0,
  "data": {
    "flags": "0x80000000000000000000000000000000",
    "free": "(rounded 200000000000000000000)",
    "frozen": 0,
    "reserved": 0,
  },
  "nonce": 2,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > from chain hrmp messages 1`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v3": [
          {
            "withdrawAsset": [
              {
                "fun": {
                  "fungible": "0x000000000000000ad78ebc5ac6200000",
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x1": {
                        "palletInstance": 10,
                      },
                    },
                    "parents": 0,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": "0x000000000000000ad78ebc5ac6200000",
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x1": {
                        "palletInstance": 10,
                      },
                    },
                    "parents": 0,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "allCounted": 1,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountKey20": {
                      "key": "0x3cd0a705a2dc65e5b1e1205896baa2be8a07c6e0",
                      "network": null,
                    },
                  },
                },
                "parents": 0,
              },
            },
          },
          {
            "setTopic": "(redacted)",
          },
        ],
      },
    ],
    "recipient": 2004,
  },
]
`;

exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > route chain xcm events 1`] = `[]`;

exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > to chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": {
        "Sibling": "(rounded 2000)",
      },
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 38000)",
        "refTime": "(rounded 7600000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > tx events 1`] = `
[
  {
    "data": {
      "assets": [
        {
          "fun": {
            "Fungible": 200000000000000000000,
          },
          "id": {
            "interior": {
              "X2": [
                {
                  "Parachain": "(rounded 2000)",
                },
                {
                  "PalletInstance": 10,
                },
              ],
            },
            "parents": 1,
          },
        },
      ],
      "dest": {
        "interior": {
          "X2": [
            {
              "Parachain": "(rounded 2000)",
            },
            {
              "AccountKey20": {
                "key": "0x3cd0a705a2dc65e5b1e1205896baa2be8a07c6e0",
                "network": null,
              },
            },
          ],
        },
        "parents": 1,
      },
      "fee": {
        "fun": {
          "Fungible": 200000000000000000000,
        },
        "id": {
          "interior": {
            "X2": [
              {
                "Parachain": "(rounded 2000)",
              },
              {
                "PalletInstance": 10,
              },
            ],
          },
          "parents": 1,
        },
      },
      "sender": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "TransferredAssets",
    "section": "xTokens",
  },
]
`;

exports[`hydration & moonbeam > moonbeam transfer DOT to hydration > balance on from chain 1`] = `
{
  "balance": 9980000000000,
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`hydration & moonbeam > moonbeam transfer DOT to hydration > balance on to chain 1`] = `
{
  "free": "(rounded 19600000000)",
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`hydration & moonbeam > moonbeam transfer DOT to hydration > from chain ump messages 1`] = `
[
  {
    "v3": [
      {
        "withdrawAsset": [
          {
            "fun": {
              "fungible": 20000000000,
            },
            "id": {
              "concrete": {
                "interior": {
                  "here": null,
                },
                "parents": 0,
              },
            },
          },
        ],
      },
      {
        "clearOrigin": null,
      },
      {
        "buyExecution": {
          "fees": {
            "fun": {
              "fungible": 10000000000,
            },
            "id": {
              "concrete": {
                "interior": {
                  "here": null,
                },
                "parents": 0,
              },
            },
          },
          "weightLimit": {
            "unlimited": null,
          },
        },
      },
      {
        "depositReserveAsset": {
          "assets": {
            "wild": {
              "allCounted": 1,
            },
          },
          "dest": {
            "interior": {
              "x1": {
                "parachain": 2034,
              },
            },
            "parents": 0,
          },
          "xcm": [
            {
              "buyExecution": {
                "fees": {
                  "fun": {
                    "fungible": 10000000000,
                  },
                  "id": {
                    "concrete": {
                      "interior": {
                        "here": null,
                      },
                      "parents": 1,
                    },
                  },
                },
                "weightLimit": {
                  "unlimited": null,
                },
              },
            },
            {
              "depositAsset": {
                "assets": {
                  "wild": {
                    "allCounted": 1,
                  },
                },
                "beneficiary": {
                  "interior": {
                    "x1": {
                      "accountId32": {
                        "id": "0xd17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69",
                        "network": null,
                      },
                    },
                  },
                  "parents": 0,
                },
              },
            },
          ],
        },
      },
    ],
  },
]
`;

exports[`hydration & moonbeam > moonbeam transfer DOT to hydration > route chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": {
        "Ump": {
          "Para": "(rounded 2000)",
        },
      },
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 7200)",
        "refTime": "(rounded 580000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`hydration & moonbeam > moonbeam transfer DOT to hydration > to chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": "Parent",
      "success": true,
      "weightUsed": {
        "proofSize": 0,
        "refTime": 500000000,
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`hydration & moonbeam > moonbeam transfer DOT to hydration > tx events 1`] = `
[
  {
    "data": {
      "outcome": {
        "Complete": {
          "used": {
            "proofSize": "(rounded 20000)",
            "refTime": "(rounded 4700000000)",
          },
        },
      },
    },
    "method": "Attempted",
    "section": "polkadotXcm",
  },
]
`;
